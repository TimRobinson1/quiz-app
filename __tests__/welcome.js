import { Text } from 'react-native';
import React from 'react';
import Welcome from '../src/components/welcome.js';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it("renders correctly", () => {
  const { navigate } = jest.fn()
  const tree = renderer.create(<Welcome />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("displays correct header", () => {
  const page = shallow(<Welcome />);
  expect(page.find('.header').node.props.children).toContain('Are you over 18?');
});

it("start button is available", () => {
  const page = shallow(<Welcome />);
  expect(page.find('.start').node.props.children).toContain('Yes');
});

it("link to drink aware website displays", () => {
  const page = shallow(<Welcome />);
  expect(page.find('.link').node.props.children).toContain('Drink Aware');
});
