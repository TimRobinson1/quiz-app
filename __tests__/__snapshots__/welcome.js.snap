// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Image
  source={1}
  style={
    Object {
      "flex": 1,
      "height": 300,
      "width": null,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(102,0,102,0.5)",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "justifyContent": "flex-end",
          "padding": 40,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        className="header"
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "white",
            "fontSize": 35,
            "marginBottom": 40,
            "textAlign": "center",
          }
        }
      >
        Are you over 18?
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 4,
          "padding": 40,
        }
      }
    >
      <View
        accessibilityComponentType="button"
        accessibilityLabel={undefined}
        accessibilityTraits="button"
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007aff",
                "fontSize": 17,
                "fontWeight": "500",
                "textAlign": "center",
              },
              null,
              Object {
                "backgroundColor": "purple",
                "borderColor": "#FFFFFF",
                "borderRadius": 15,
                "borderWidth": 1,
                "color": "#FFFFFF",
                "fontSize": 40,
                "height": 50,
                "overflow": "hidden",
              },
              null,
            ]
          }
        >
          Yes
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(102,0,102,0.5)",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      className="link"
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "#538EED",
          "fontSize": 20,
          "textAlign": "center",
          "textDecorationLine": "underline",
        }
      }
    >
      Drink Aware
    </Text>
  </View>
</Image>
`;
